de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ArchiveIncomingProjectNegotiation.java

```diff
@@ -108,6 +108,7 @@ protected void transfer(IProgressMonitor monitor,
     @Override
     protected void cleanup(IProgressMonitor monitor,
         Map<String, IProject> projectMapping) {
+        addProjectsToSession(projectMapping);
```

Comment:
Why is this called in the cleanup and not at the end of the `transfer(...)` method?

It should basically be the same order (only `checkCancellation(...)` would be called afterwards instead of beforehand). The only big difference would be that with the current implementation, the `addprojectToSession(...)` call is located in the `finally` block, meaning it also gets executed when the incoming project negotiation fails. This is probably the wrong behavior as it does not make much sense to add the "shared" resources to the session when the negotiation/resource sharing failed.

Reply:
You are right, it should move into transfer. Somehow I thought cleanup is called right after transfer, but even then it should move there.